src/test/java/com/limechain/runtime/RuntimeWasmTest.java

```diff
@@ -11,8 +10,13 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
-@Disabled("This is an integration test")
+@Disabled("This is an integration test.")
 class RuntimeWasmTest {
+    // TODO:
+    //  This wasm is not a valid runtime since it doesn't obey the runtime API (doesn't export `Core_version`).
+    //  This test suite needs total refactoring, since our Runtime class models a wasm module obeying the contract
+    //  to call it a "Runtime", i.e. the entire runtime API, memory exports, etc. as per spec.
+    //  Currently, we don't have a smaller class modelling "an instance of an arbitrary wasm module"
```

Comment:
That's an important consideration, although it arises from this practically deprecated test.

I'd like to trace somewhere that we've made a conscious design decision on that - do we actually need appropriate infrastructure for things less than "a valid runtime wasm" as per spec?

This test, for example, currently fails because the provided source doesn't export `Core_version`, which we call on construction in order to tie it.

We might EITHER decide that the test is not necessary (although it points to an understandable desire to be able to test something smaller in isolation) OR decide that this test's desires are respectable, in which case we'd have to at least not rely on calling `Core_version` on instantiation... (to past this test, at least), which then means that `RuntimeBuilder` is not actually a builder for runtimes only, but for any wasm module (breaks the spec contract), which then evolves into us introducing separate logic for handling wasm instances, which are not runtimes. We don't have that yet. The current design is concerned with building valid runtimes only.

Reply:
I think `hello_world.wat` is not the proper wasm for this test. Valid runtime wasms should be standard wasm we use for any kind of runtime tests(unless the specific unit tests requires that).

Reply:
We should get rid of `hello_world.wasm`. `runtime.wasm` could be used instead - it is currently used for the MiscHostApi's getRuntimeVersion test

Reply:
hello_world is created from a simple rust file printing hello world. It's not made to be a polkadot runtime but rather used because of it's small size for testing the 2 exports that every wasm file exports - data end and heap base.

Reply:
So it doesn't require any of the imports from the Host API, right? If that's the case, isn't this a test of `wasmer-java` instead, because I see no other logic being tested on our side. If that's the case, I'd remove it completely.

Reply:
Yeah, feel free to remove it

Reply:
Fixed in [9778418](https://github.com/LimeChain/Fruzhin/pull/456/commits/9778418c89b310d2b68990c3a0cf570651e1175f).